day4/cards.go

```diff
@@ -0,0 +1,51 @@
+package day4
+
+import (
+	"bufio"
+	"math"
+	"os"
+	"strings"
+)
+
+// CountMatches counts the total number of matches for a given "Card".
+func CountMatches(cardData string) int {
+	winnersAndReveals := strings.Split(strings.Split(cardData, ": ")[1], " | ")
+
+	revealed := make(map[string]bool)
+
+	for _, v := range strings.Split(winnersAndReveals[1], " ") {
+		if v != "" {
+			revealed[v] = true
+		}
+	}
+
+	matches := 0
+
+	for _, v := range strings.Split(winnersAndReveals[0], " ") {
+		if v == "" {
+			continue
+		}
+		if revealed[v] {
+			matches++
+		}
+	}
```

Comment:
Unlike my C# implementation, I'm doing as little work here as possible. I'm splitting the strings on the spaces, creating a map for the 'revealed' numbers so that I can get O(1) lookup when I get to the 'winner' numbers. Then I just check each winner number to see if it is in the 'revealed' map, incrementing 'matches' whenever I find a match. Pretty quick.